core/src/main/java/kafka/server/share/SharePartitionManager.java

```diff
@@ -250,13 +250,39 @@ public CompletableFuture<Map<TopicIdPartition, ShareAcknowledgeResponseData.Part
         String groupId,
         Map<TopicIdPartition, List<ShareAcknowledgementBatch>> acknowledgeTopics
     ) {
-        log.trace("Acknowledge request for topicIdPartitions: {} with groupId: {}",
+        log.debug("Acknowledge request for topicIdPartitions: {} with groupId: {}",
             acknowledgeTopics.keySet(), groupId);
+        Map<TopicIdPartition, CompletableFuture<Errors>> futures = new HashMap<>();
+        acknowledgeTopics.forEach((topicIdPartition, acknowledgePartitionBatches) -> {
+            SharePartition sharePartition = partitionCacheMap.get(sharePartitionKey(groupId, topicIdPartition));
+            if (sharePartition != null) {
+                synchronized (sharePartition) {
+                    CompletableFuture<Errors> future = sharePartition.acknowledge(memberId, acknowledgePartitionBatches).thenApply(throwable -> {
+                        if (throwable.isPresent()) {
+                            return Errors.forException(throwable.get());
+                        } else {
```

Comment:
nit: Is this else block required? The `if` already returns. Hence we can just write the code without `else` block.